### PR TITLE
David kenzy/scrollable calendar with functionality

### PR DESCRIPTION
**Pull request by ludavidca**
## Notion ticket link

[Make Calendar Scrollable](https://www.notion.so/uwblueprintexecs/Make-calendar-scrollable-13a1d59b855b4700a24802f67d568890)

## Implementation description

- Calendar
  - Implemented scrollable calendar with infinite scroll (3 years forward, 1 year back)
  - Added notes to schema.prisma and submission to input form
  - Connected the months when scrolling
  - Implemented the "today" button (auto scrolls to today)
  - Created UI to display each individual event with the correct location and class type
  - Implemented highlighting of today’s date
  - Implemented dynamic rendering of months based on displayed dates 

- Input form
  - Updated the input form component to reflect new design
  - Implemented input form appearance with today’s date when the "declare absence" button is clicked
  - Implemented input form appearance with the corresponding date when a date block is clicked
  - Implemented background darkening when the input form appears
  - Created the Google Drive upload component and connected API routes for the input form
  - Created a mini calendar form to select dates
  - Implemented successful message
  - Implemented validation and alerts when fields are not present

***NO ERROR HANDLING YET DUE TO DROPDOWN PR**

- API 
  - Created API Route to get all absence data from the calendar with the correct class type and locations
  - Created API Routes for adding all data fields to absence table
  - Integrated prisma.ts file
 

## Steps to test

- Test Calendar:
  - Scroll through the calendar to verify that infinite scroll works both forwards and backward (2 years in each direction) and that months render/update properly
  - Verify that the months stay properly connected/aligned when scrolling
  - Click the "today" button and verify that it smoothly scrolls to the current date
  - Verify that today's date is visually highlighted
  - Verify that events display correctly with:
    - Proper location information
    - Correct class type
    - Accurate dates

- Test Input Form:
  - Via the "declare absence" button:
    - Verify form opens with today's date pre-selected
  - Via a date block click:
    - Verify form opens with clicked date pre-selected
    
- Test form validation:
  - Submit an empty form to verify validation alerts
  - Verify each required field individually

**DO NOT TEST FOR INVALID Teacher ID error as this should not be an issue once dropdown is implemented**

- Test Google Drive upload:
  - Upload various file types to verify accepted formats
  - Test file size limits
  - Verify successful upload notification

- Test mini calendar:
  - Verify date selection works
  - Verify navigation between months
  - Verify that the selected date is reflected in the main input form

- Test form submission:
  - Submit valid data and verify the success message
  - Verify that submitted data appears correctly in the calendar
  - Verify all fields are properly saved to the database

- Test form dismissal:
  - Click outside of the form to verify it closes
  - Verify background returns to normal
  - Verify unsaved data is handled appropriately

- Test API:
  - Verify GET requests return correct absence data:
  - Verify all events are displayed
  - Verify class types are accurate
  - Verify locations are correct
  - Verify POST requests for absence submission:
  - Verify all data fields are properly saved
  - Verify error handling for invalid data
  - Verify successful database updates
  - Verify Google Drive link gets inputted into databases

## What should reviewers focus on?
- Proper input validation 
- Proper input updates/reflection in the calendar/database
- Code readability and if sections/methods make sense
- General code style/best practices

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: prisma/seed/.snaplet/dataModel.json
Comment (ChinemeremChigbo) [inline prisma/seed/.snaplet/dataModel.json:1]: Revert

File: package-lock.json
Comment (ChinemeremChigbo) [inline package-lock.json:1]: We shouldn't delete the package-lock.json

**Comment by VJagiasi** [comment]
> * failed submission messages

remove this from the ReadMe for testing, this hasn't been implemented and will be done in the dropdown pr

**Comment by VJagiasi** [comment]
> * timing

change this to accurate date since timing would also include the exact time

**Comment by VJagiasi** [comment]
> * Test Google Drive upload:

the file gets uploaded to the drive even if I don't submit the input form - fix it so that it get's uploaded only when we submit the absence

File: prisma/schema.prisma [hunk 1]
```diff
@@ -20,6 +20,7 @@ model Absence {
   substituteTeacher   User?    @relation("substituteTeacher", fields: [substituteTeacherId], references: [id])
   location            Location @relation(fields: [locationId], references: [id])
   subject             Subject  @relation(fields: [subjectId], references: [id])
+  notes               String?
```
Comment (ChinemeremChigbo) [inline prisma/schema.prisma:23]: You might need to run a migration since you modified the schema

File: prisma/seed/.snaplet/dataModel.json
Comment (ChinemeremChigbo) [inline prisma/seed/.snaplet/dataModel.json:1]: Please set your default JSON formatter to `Prettier - Code formatter` rather `JSON Language Features`

File: prisma/schema.prisma (hunk 1 above)
Comment (ludavidca) [reply prisma/schema.prisma:23]: Completed

**Pull request closed without merging**